central/complianceoperator/v2/report/manager/watcher/scanwatcher.go

```diff
@@ -281,13 +302,53 @@ func (s *scanWatcherImpl) handleScan(scan *storage.ComplianceOperatorScanV2) err
 		}
 		s.totalChecks = numChecks
 	}
+
+	s.resultsLock.Lock()
+	defer s.resultsLock.Unlock()
+
 	if s.scanResults.Scan == nil {
 		s.scanResults.Scan = scan
 	}
+	// If we received a newer timestamp we need to reset the watcher.
+	if protocompat.CompareTimestamps(s.lastStartedTime, scan.GetLastStartedTime()) < 0 {
+		s.lastStartedTime = scan.GetLastStartedTime()
+		s.scanResults.Scan = scan
+		s.scanResults.CheckResults = set.NewStringSet()
+		s.timeout.Reset()
+	}
```

Comment:
Okay, so at this point, we forget everything from the scan run with older timestamp, right?

Reply:
Yeah. Two Scans with the same ID cannot run at the same time. So, if we receive a newer ID while the older one is still running, we simply reset the watcher.

Reply:
> Two Scans with the same ID cannot run at the same time.

I know that this is how the business logic works, but Sensor can receive messages for the new scan while the previous one has not finished yet, so that may look like the scans running in parallel.

But your explanation matches with my understanding of that part, so all is good.